addons/pos_online_payment/controllers/payment_portal.py

```diff
@@ -0,0 +1,299 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import urllib.parse
+
+from odoo import _, http, tools
+from odoo.http import request
+from odoo.exceptions import AccessError, ValidationError, UserError, MissingError
+from odoo.addons.payment.controllers.post_processing import PaymentPostProcessing
+from odoo.addons.payment.controllers import portal as payment_portal
+
+
+class PaymentPortal(payment_portal.PaymentPortal):
+
+    def _check_order_access(self, pos_order_id, access_token):
+        try:
+            order_sudo = self._document_check_access(
+                'pos.order', self._cast_as_int(pos_order_id), access_token)
+        except:
+            raise AccessError(
+                _("The provided order or access token is invalid."))
+
+        if order_sudo.state == "cancel":
+            raise ValidationError(_("The order has been canceled."))
+        return order_sudo
+
+    def _ensure_session_open(self, pos_order_sudo):
+        if pos_order_sudo.session_id.state != 'opened':
+            raise AccessError(_("The POS session is not opened."))
+
+    def _get_partner_sudo(self, user_sudo):
+        partner_sudo = user_sudo.partner_id
+        if not partner_sudo and user_sudo._is_public():
+            partner_sudo = self.env.ref('base.public_user')
+        return partner_sudo
+
+    @staticmethod
+    def _redirect_login(request):
+        return request.redirect(f'/web/login?redirect={urllib.parse.quote(request.httprequest.full_path)}')
+
+    @staticmethod
+    def _is_valid_currency(currency):
+        return currency and currency.active
```

Comment:
Yes, it is **currently** just that. If tomorrow we need to add a check, it will be really easy: just one method to change.
Added to that, the method highlights and guarantees that the currency will be checked the same way in pos_order_pay and pos_order_pay_transaction.
So I disagree with you, this function is not useless.